app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusHandlerThread.java

```diff
@@ -152,30 +160,56 @@ public void quit() {
     }
 
     public void sync() {
+        LOG.info("HPlus: Starting data synchronization");
+
         mGetSleepTime.setTimeInMillis(0);
         mGetDaySlotsTime.setTimeInMillis(0);
         mGetDaySummaryTime.setTimeInMillis(0);
         mLastSleepDayReceived.setTimeInMillis(0);
+        mHelloTime = GregorianCalendar.getInstance();
+        mHelloTime.add(Calendar.SECOND, HELLO_PERIOD);
 
         mSlotsInitialSync = true;
         mLastSlotReceived = -1;
         mLastSlotRequested = 0;
         mCurrentDaySlot = null;
         mDaySlotRecords.clear();
 
-        TransactionBuilder builder = new TransactionBuilder("startSyncDayStats");
+        try {
+            if(!mHPlusSupport.isConnected())
+                mHPlusSupport.connect();
 
-        builder.write(mHPlusSupport.ctrlCharacteristic, new byte[]{HPlusConstants.CMD_GET_DEVICE_ID});
-        builder.write(mHPlusSupport.ctrlCharacteristic, new byte[]{HPlusConstants.CMD_GET_VERSION});
-        builder.write(mHPlusSupport.ctrlCharacteristic, new byte[]{HPlusConstants.CMD_GET_CURR_DATA});
+            TransactionBuilder builder = new TransactionBuilder("startSyncDayStats");
```

Comment:
That was my initial approach but it crashed with an AssertionError "Attempt to repark", which I could not solve.